include/obstack.h

```diff
@@ -0,0 +1,459 @@
+ *   only allocated amount. The non-standard obstack_finish_norealloc can be
+ *   used if you want the standard behavior for ever reason.
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *
+ * Returned Value:
+ *   Permanent address to the object.
+ *
+ ****************************************************************************/
+
+FAR void *obstack_finish (FAR struct obstack *h);
+
+/****************************************************************************
+ * Name: obstack_finish_norealloc
+ *
+ * Description:
+ *   Finish growing object and receive address to it without reallocating
+ *   buffer to fit the object (keeping space for more growth).
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *
+ * Returned Value:
+ *   Permanent address to the object.
+ *
+ ****************************************************************************/
+
+FAR void *obstack_finish_norealloc (FAR struct obstack *h);
+
+/****************************************************************************
+ * Name: obstack_object_size
+ *
+ * Description:
+ *   Calculate the size of the currently growing object.
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *
+ * Returned Value:
+ *   Size of the object.
+ *
+ ****************************************************************************/
+
+size_t obstack_object_size (FAR struct obstack *h);
+
+/****************************************************************************
+ * Name: obstack_room
+ *
+ * Description:
+ *   Calculate the number of bytes available for growth before reallocation
+ *   is required.
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *
+ * Returned Value:
+ *   Number of free bytes.
+ *
+ ****************************************************************************/
+
+size_t obstack_room (FAR struct obstack *h);
+
+/****************************************************************************
+ * Name: obstack_printf
+ *
+ * Description:
+ *   This is similar to the asprintf except it uses obstack to allocate
+ *   string on. The characters are written onto the end of the currently
+ *   growing object and terminated by null byte.
+ *
+ *   This function is defined in stdio.h in GlibC. There is no definition
+ *   that would be in stdio.h required for these here and thus it is easier
+ *   to just keep these functions here as user has to include obstack anyway
+ *   to get the full functionality.
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   fmt: format string with its format inputs followed.
+ *
+ * Returned Value:
+ *   Number of characters added to the obstack excluding the null byte.
+ *
+ ****************************************************************************/
+
+int obstack_printf(FAR struct obstack *h, FAR const char *fmt, ...);
```

Comment:
add printflike

Reply:
I missed that there is some macro for it. :+1: